api/scpca_portal/test/test_metadata_file.py

```diff
@@ -68,13 +75,24 @@ def test_write_metadata_dicts_successful_write(self):
         self.assertTrue(os.path.exists(self.dummy_output_path))
 
     def test_write_metadata_dicts_read_write_values_match(self):
+        expected_output = self.dummy_list_of_dicts.copy()
+        expected_output[len(expected_output) - 1]["capital"] = common.NA
```

Comment:
For a test like this, it is best not to do this programmatically. Just create a new object with the expected output.

I would recommend making the following changes:

- shorten the list of dummy data to 3 items
- put them out of order in the dummy data with a comment saying that this is intentional
- create a new object in this test that has the expected output and compare that against the result

Reply:
I've applied your feedback, pls take a look at bda09e5055ad505e69a8570b33100e81f5f60e49, thank you David!

Reply:
FYI: I've made the minor change to dummy data ID here fd2a8cf248637d9da3521ccdb0b16c44cb4f5fd7.